pkg/operator/ceph/cluster/cluster_cleanup.go

```diff
@@ -0,0 +1,185 @@
+	for node := range cephHosts {
+		logger.Infof("starting clean up job on node %q", node)
+		jobName := k8sutil.TruncateNodeName("cluster-cleanup-job-%s", node)
+		podSpec := c.cleanUpJobTemplateSpec(cluster)
+		podSpec.Spec.NodeSelector = map[string]string{v1.LabelHostname: node}
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      jobName,
+				Namespace: cluster.Namespace,
+				Labels:    controller.AppLabels(appName, cluster.Namespace),
+			},
+			Spec: batch.JobSpec{
+				Template: podSpec,
+			},
+		}
+
+		if err := k8sutil.RunReplaceableJob(c.context.Clientset, job, true); err != nil {
+			logger.Error("failed to run cluster clean up job on node %q. %+v", node, err)
+		}
+	}
+}
+
+func (c *ClusterController) cleanUpJobContainer(cluster *cephv1.CephCluster) v1.Container {
+	privileged := true
+	volumeMounts := []v1.VolumeMount{}
+	hostPathVolumeMount := v1.VolumeMount{Name: volumnName, MountPath: cluster.Spec.DataDirHostPath}
+	volumeMounts = append(volumeMounts, hostPathVolumeMount)
+	envVars := []v1.EnvVar{
+		{Name: dataDirHostPath, Value: fmt.Sprintf("%s/%s", cluster.Spec.DataDirHostPath, cluster.Namespace)},
+	}
+
+	container := v1.Container{
+		Name:  "clean-up",
+		Image: c.rookImage,
+		SecurityContext: &v1.SecurityContext{
+			Privileged: &privileged,
+		},
+		VolumeMounts: volumeMounts,
+		Env:          envVars,
+		Args:         []string{"ceph", "clean"}}
+
+	return container
+}
+
+func (c *ClusterController) cleanUpJobTemplateSpec(cluster *cephv1.CephCluster) v1.PodTemplateSpec {
+	volumes := []v1.Volume{}
+	hostPathVolume := v1.Volume{Name: volumnName, VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: cluster.Spec.DataDirHostPath}}}
+	volumes = append(volumes, hostPathVolume)
+
+	podSpec := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: appName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				c.cleanUpJobContainer(cluster),
+			},
+			Volumes:       volumes,
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+
+	return podSpec
+}
+
+func (c *ClusterController) waitForCephDaemonCleanUp(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+	for {
+		cephHosts, err := c.getCephHosts(cluster.Namespace)
+		if err != nil {
+			return errors.Wrap(err, "failed to list ceph daemon nodes")
+		}
+
+		if len(cephHosts) == 0 {
+			logger.Info("all ceph daemons cleaned up")
+			break
+		}
+
+		logger.Infof("waiting for ceph daemons in cluster %s to be cleaned up. Retrying in %s",
+			cluster.Namespace, retryInterval.String())
+		for host, pods := range cephHosts {
+			logger.Infof("hostname %q. Pending ceph daemon count: %d", host, len(pods))
+		}
+		<-time.After(retryInterval)
+	}
+
+	return nil
+}
+
+func (c *ClusterController) getCephHosts(namespace string) (map[string][]v1.Pod, error) {
+	cephPods := make([]v1.Pod, 0)
+	cephAppNames := []string{mon.AppName, mgr.AppName, osd.AppName, object.AppName, mds.AppName, rbd.AppName}
```

Comment:
We just need mon and osd, don't we?

Reply:
All the daemons use the host path for the crash collector, right?

Reply:
Oh yeah forgot about that as well as the logs!